compiler_opt/rl/data_reader.py

```diff
@@ -24,16 +24,21 @@
 from compiler_opt.rl import constant
 
 
-def _get_policy_info_parsing_dict(agent_name, action_spec):
+def _get_policy_info_parsing_dict(agent_name, action_spec, is_greedy):
   """Function to get parsing dict for policy info."""
-  if agent_name == constant.AgentName.PPO:
+  if agent_name in (constant.AgentName.PPO,
+                    constant.AgentName.PPO_DISTRIBUTED) and not is_greedy:
     if tensor_spec.is_discrete(action_spec):
-      return {
+      parsing_dict = {
           'CategoricalProjectionNetwork_logits':
               tf.io.FixedLenSequenceFeature(
                   shape=(action_spec.maximum - action_spec.minimum + 1),
-                  dtype=tf.float32)
+                  dtype=tf.float32),
       }
+      if agent_name == constant.AgentName.PPO_DISTRIBUTED:
+        parsing_dict['value_prediction'] = tf.io.FixedLenSequenceFeature(
```

Comment:
this causes some code discrepancy between distributed and non-distributed training that we'd want to avoid. 

is it possible to remove it?

Reply:
This is because we set `compute_value_and_advantage_in_train=False` for distributed training. TPUs use mini batches when learning, and that parameter must be False if mini batches are enabled. See: https://github.com/tensorflow/agents/blob/master/tf_agents/agents/ppo/ppo_agent.py#L273

Reply:
we are using mini batches in the non-distributed training as well, i don't think it's really a hard requirement

Reply:
For distributed training, we are using the tf_agents `PPOLearner` as it is optimized for tpu training (no context switches between device and host, etc). The requirement for `compute_value_and_advantage_in_train=False` comes from this line (https://github.com/tensorflow/agents/blob/10ad6b054bb508adc5a8543ea4849d21f6cc6304/tf_agents/train/ppo_learner.py#L176) which isn't an issue in the non-distributed training as we have our own custom trainer.

Reply:
it seems that you patched the original PPOLearner implementation from PR #184 . Is it possible to remove this constraint?

Reply:
I spent some time this morning trying to do this and unfortunately it seems that although it is theoretically possible, it would require a lot of work. It seems like there is an implicit assumption in tf_agents that in the distributed setting, the value computation happens during data collection. I got around this in some spots, but then ran into other issues around the data pipeline. I think the easiest solution would be to include this bit of code in data_reader (I don't think it's too bad, just two extra `if` statements that are clearly intended for the distributed agent)